src/move_basic.cpp

```diff
@@ -799,8 +802,28 @@ bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
                                                             forwardRight);
         }
 
+	// Control to reference pose 
+	double K = 1.5;
+	double MOTOR_CONTROLLER_REACTIVITY = 0.02; // Changing this variable affects the reference pose accuracy, but it also regulates the deccelaration time - TODO: If higher or equal than 0.4 throws exception
+	if ((Pvelocity < MOTOR_CONTROLLER_REACTIVITY) && first) { 
+		if (distRemaining < linearTolerance) { // Checks if in tolerance range
+			sendCmd(0, 0);
+        		ROS_INFO("MoveBasic: Done linear, error %f, %f meters", remaining.x(), remaining.y());
+			return true;
+		}
+		else {
+			sendCmd(0, 0);
+			abortGoal("MoveBasic: Aborting due to linear error");
+			return false;
+		}
+	}
+	else {
+		Pvelocity = K * distRemaining;
+		first = true;
+	}
+
         double velocity = std::max(minLinearVelocity,
```

Comment:
This whole calcuation is getting pretty large and confusing to read, any way that we can clean it up?